### PR TITLE
fix(engine): populate ContentNode.loop for all with_* loop variants

### PR DESCRIPTION
**Pull request by cidrblock**
## Summary

- Fix `GraphBuilder._build_task` to populate `ContentNode.loop` for all 16+ `with_*` loop constructs, not just `loop` and `with_items`
- Unblocks GraphRule ports that need to detect looping tasks (e.g., a future L061 graph port)

## Changes

- **`src/apme_engine/engine/content_graph.py`** — Replace hardcoded `options.get("loop") or options.get("with_items")` with a generic `options` dict scan that catches any key starting with `with_`. Modern `loop:` still takes precedence via short-circuit evaluation.
- **`tests/test_content_graph.py`** — Add `TestGraphBuilderLoopVariants` with 16 tests: modern `loop:`, `with_items`, `with_dict`, `with_sequence`, `with_fileglob`, `with_subelements`, `with_nested`, `with_together`, `with_flattened`, `with_indexed_items`, `with_random_choice`, `with_cartesian`, no-loop (`None`), `loop:` precedence over `with_*`, plus two falsy-loop edge cases documenting that `loop: []` is treated as absent (matching Ansible semantics where zero-iteration loops are no-ops).

## Context

Discovered while reviewing PR #158 (L061 set_fact+loop+when rule). The existing `ContentNode.loop` field silently returned `None` for tasks using `with_dict`, `with_sequence`, and 12 other loop constructs — making the field unreliable for any GraphRule that checks `node.loop`. The OPA payload builder (`graph_opa_payload.py`) already independently re-extracted `with_*` keys from `node.options` as a workaround; this fix makes `node.loop` trustworthy at the source.

**Existing rules affected:** `L049_loop_var_prefix_graph.py` uses `node.loop is not None` in its `match()` — this fix is strictly positive for L049, which was silently skipping tasks using `with_dict`, `with_sequence`, etc. The data shape (raw YAML value from options) is unchanged, so L049's behavior improves without code changes.

The `or`-based truthiness semantics (falsy values like `loop: []` fall through) are pre-existing from the original code and intentionally preserved: in Ansible, `loop: []` is a zero-iteration no-op, so treating it as absent is correct behavior.

## Test plan

- [x] `prek run --all-files` passes
- [x] All 89 graph-related tests pass (87 existing + 2 new falsy-loop edge cases)
- [x] 16 new loop variant tests pass
- [x] L049 graph rule behavior verified (strictly improved — now matches `with_*` tasks)

### CODE REVIEW

File: tests/test_content_graph.py [hunk 1]
```diff
@@ -266,6 +268,154 @@ def test_definitions_with_empty_mappings(self) -> None:
...
+
+    def test_empty_loop_treated_as_no_loop(self) -> None:
+        """Falsy loop values (empty list/dict) are treated as no loop.
+
+        In Ansible, ``loop: []`` is a no-op (zero iterations), so
+        treating it as absent is correct behavior.
+        """
+        node = _build_task_node_with_options({"loop": []})
+        assert node.loop is None
+
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply tests/test_content_graph.py]: Good catch. Split the test into `test_empty_list_loop_treated_as_no_loop` and `test_empty_dict_loop_treated_as_no_loop` so both cases are explicitly covered. Fixed in ce16351.
